addons/website_profile/models/res_users.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import hashlib
+
+from datetime import datetime
+from werkzeug import urls
+from odoo import api, models
+
+VALIDATION_KARMA_GAIN = 3
+
+
+class Users(models.Model):
+    _inherit = 'res.users'
+
+    @api.model
+    def _generate_profile_token(self, user_id, email):
+        """Return a token for email validation. This token is valid for the day
+        and is a hash based on a (secret) uuid generated by the forum module,
+        the user_id, the email and currently the day (to be updated if necessary). """
+        profile_uuid = self.env['ir.config_parameter'].sudo().get_param('website_profile.uuid')
+        return hashlib.sha256((u'%s-%s-%s-%s' % (
+            datetime.now().replace(hour=0, minute=0, second=0, microsecond=0),
+            profile_uuid,
+            user_id,
+            email
+        )).encode('utf-8')).hexdigest()
+
+
+    @api.one
+    def _send_profile_validation_email(self, **kwargs):
+        if not self.email:
+            return False
+        token = self._generate_profile_token(self.id, self.email)
+        activation_template = self.env.ref('website_profile.validation_email')
+        if activation_template:
+            params = {
+                'token': token,
+                'id': self.id,
+                'email': self.email}
+            params.update(kwargs)
+            base_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
+            token_url = base_url + '/profile/validate_email?%s' % urls.url_encode(params)
+            with self._cr.savepoint():
+                activation_template.sudo().with_context(token_url=token_url).send_mail(
+                    self.id, force_send=True, raise_exception=True)
+        return True
+
+    @api.one
+    def process_profile_validation_token(self, token, email, context=None, **kwargs):
```

Comment:
While we are at it; remove the `api.one` to make it `multi` and add a `self.ensure_one()`. Carrefull the return will not be a list anymore. Also, make it private. And remove the context argument, and the kwargs.
This method was definitely badly written ...